src/showingpreviously/cinemas/odeon.py

```diff
@@ -0,0 +1,146 @@
+import json
+import re
+
+from selenium import webdriver
+from datetime import datetime, timedelta
+from typing import Tuple, Iterator
+import showingpreviously.requests as requests
+from showingpreviously.model import ChainArchiver, CinemaArchiverException, Chain, Cinema, Screen, Film, Showing
+
+CINEMAS_URL = 'https://vwc.odeon.co.uk/WSVistaWebClient/ocapi/v1/browsing/master-data/sites'
+FILMS_URL = 'https://vwc.odeon.co.uk/WSVistaWebClient/ocapi/v1/browsing/master-data/films'
+SHOWINGS_URL = 'https://vwc.odeon.co.uk/WSVistaWebClient/ocapi/v1/browsing/master-data/showtimes/business-date/{date}'
+
+JWT_TOKEN = None
+DAYS_AHEAD = 2
+CHAIN = Chain('Odeon')
+
+
+def get_jwt_token() -> str:
+    global JWT_TOKEN
+    if JWT_TOKEN is None:
```

Comment:
My concern is that, in a production environment, this class may be loaded once and then kept loaded. In this case there is the danger that the token will become stale. My thoughts would be to either:
1. Fetch the token every time the get_showings() method is called.
2. Have a way of checking that the token is expired.

Reply:
I've redone all of this, and now it isn't cached so it can't become stale.